internal/ethapi/api.go

```diff
@@ -1134,7 +1134,9 @@ func (s *BlockChainAPI) Call(ctx context.Context, args TransactionArgs, blockNrO
 	return result.Return(), result.Err
 }
 
-func DoEstimateGas(ctx context.Context, b Backend, args TransactionArgs, blockNrOrHash rpc.BlockNumberOrHash, gasCap uint64) (hexutil.Uint64, error) {
+// DoEstimateGas estimates the gas needed to execute a specific transaction.
+// Note, the flag noBaseFee indicating whether to skip the base fee check
+func DoEstimateGas(ctx context.Context, b Backend, args TransactionArgs, blockNrOrHash rpc.BlockNumberOrHash, gasCap uint64, noBaseFee bool) (hexutil.Uint64, error) {
```

Comment:
If we want to do a gas estimation, why would we want to avoid the basefee calculation? 

Couldn't we always use `false` in `DoEstimateGas`?

Reply:
Erm, obviously I'm missing something. Please enlighten me, I don't get it why some places need to `false` and some `true`